src/lib/mdns/minimal/ResponseSender.cpp

```diff
@@ -0,0 +1,159 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ResponseSender.h"
+
+#include "QueryReplyFilter.h"
+
+#include <support/ReturnMacros.h>
+
+#define RETURN_IF_ERROR(err)                                                                                                       \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        if (err)                                                                                                                   \
+        {                                                                                                                          \
+            return;                                                                                                                \
+        }                                                                                                                          \
+    } while (false)
+
+namespace mdns {
+namespace Minimal {
+
+namespace {
+
+constexpr uint16_t kMdnsStandardPort = 5353;
+constexpr uint16_t kPacketSizeBytes  = 512;
+
+} // namespace
+
+CHIP_ERROR ResponseSender::Respond(const QueryData & query, const chip::Inet::IPPacketInfo * querySource)
+{
+    mSendError = CHIP_NO_ERROR;
+
+    mCurrentSource = querySource;
+    mSendUnicast   = query.GetUnicastAnswer();
+
+    mResponder->ResetAdditionals();
+
+    mCurrentResourceType = ResourceType::kAnswer; // direct answer
+    QueryReplyFilter filter(query);
+    for (auto it = mResponder->begin(); it != mResponder->end(); it++)
+    {
+        Responder * responder = it->responder;
+
+        if (!filter.SendAnswer(responder->GetQType(), responder->GetQClass(), responder->GetQName()))
+        {
+            continue;
+        }
+
+        responder->AddAllResponses(querySource, this);
+        ReturnErrorOnFailure(mSendError);
+
+        mResponder->MarkAdditionalReplyesFor(it);
+    }
+
+    mCurrentResourceType = ResourceType::kAdditional; // Additional parts
+    filter.SetIgnoreNameMatch(true);
+    for (auto it = mResponder->additional_begin(); it != mResponder->additional_end(); it++)
+    {
+        Responder * responder = it->responder;
+
+        if (!filter.SendAnswer(responder->GetQType(), responder->GetQClass(), responder->GetQName()))
```

Comment:
Renamed to 'Accept' since this is a 'Filter'. 

I also find this a bit dodgy: we have a filter virtual base class, with a single instance of QueryFilter that is a single simple method. At least the virtual base should go away. Future PR though.